### PR TITLE
npm version bug fix

### LINKED ISSUE #120: npm2deb '@' package name  are failing
`npm2deb create @ava/write-file-atomic` will fail. For more details please look at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=909101

### PR DESCRIPTION
**Pull request by juggernaut451**
fix #120

### CODE REVIEW

**Comment by pravi** [comment]
@juggernaut451 try npm2deb create @types/jquery@2.0.49 which fails.

**Comment by juggernaut451** [comment]
can you please test it now.

File: npm2deb/utils.py [hunk 1]
```diff
@@ -138,6 +138,9 @@ def create_dir(dir):
 
 def parse_name(name):
     parts = name.partition('@')
+    if parts[0] == '':
```
Comment (shanavas786) [inline npm2deb/utils.py]: Thanks for the fix.
We could do the same by using `rpartition` method instead of `partition`.

Please check and update.

**Comment by shanavas786** [comment]
@pravi @juggernaut451  can you test `@types/jquery` without version

**Comment by shanavas786** [comment]
Also, it would be great if you can add a couple of unit tests for parse name function

**Comment by pravi** [comment]
Working fine.

```
pravi@nishumbha:/tmp$ npm2deb create @types/jquery

Downloading source tarball file using debian/watch file...
uscan: Newest version of node-types-jquery on remote site is 3.3.10, specified download version is 3.3.10

Creating debian source package...
uupdate: debian/source/format is "3.0 (quilt)".
uupdate: Auto-generating node-types-jquery_3.3.10-1.debian.tar.xz
uupdate: -> Use existing node-types-jquery_3.3.10-1.debian.tar.xz
dpkg-source: info: extracting node-types-jquery in node-types-jquery-3.3.10
dpkg-source: info: unpacking node-types-jquery_3.3.10.orig.tar.gz
dpkg-source: info: unpacking node-types-jquery_3.3.10-1.debian.tar.xz
uupdate: Remember: Your current directory is changed back to the old source tree!
uupdate: Do a "cd ../node-types-jquery-3.3.10" to see the new source tree and
    edit it to be nice Debianized source.

Building the binary package
dpkg-source: info: using source format '3.0 (quilt)'
dpkg-source: info: building node-types-jquery using existing ./node-types-jquery_3.3.10.orig.tar.gz
dpkg-source: info: building node-types-jquery in node-types-jquery_3.3.10-1.debian.tar.xz
dpkg-source: info: building node-types-jquery in node-types-jquery_3.3.10-1.dsc
dpkg-buildpackage: info: source package node-types-jquery
dpkg-buildpackage: info: source version 3.3.10-1
dpkg-buildpackage: info: source distribution UNRELEASED
dpkg-buildpackage: info: source changed by Pirate Praveen <praveen@debian.org>
dpkg-buildpackage: info: host architecture amd64
 dpkg-source --before-build node-types-jquery-3.3.10
 fakeroot debian/rules clean
dh clean
   dh_clean
 dpkg-source -b node-types-jquery-3.3.10
dpkg-source: info: using source format '3.0 (quilt)'
dpkg-source: info: building node-types-jquery using existing ./node-types-jquery_3.3.10.orig.tar.gz
dpkg-source: info: building node-types-jquery in node-types-jquery_3.3.10-1.debian.tar.xz
dpkg-source: info: building node-types-jquery in node-types-jquery_3.3.10-1.dsc
 debian/rules build
dh build
   dh_update_autotools_config
   dh_autoreconf
   create-stamp debian/debhelper-build-stamp
 fakeroot debian/rules binary
dh binary
   dh_testroot
   dh_prep
   dh_install
   dh_installdocs
   dh_installchangelogs
   dh_perl
   dh_link
   dh_strip_nondeterminism
   dh_compress
   dh_fixperms
   dh_missing
   dh_installdeb
   dh_gencontrol
   dh_md5sums
   dh_builddeb
dpkg-deb: building package 'node-types-jquery' in '../node-types-jquery_3.3.10-1_all.deb'.
 dpkg-genbuildinfo
 dpkg-genchanges  >../node-types-jquery_3.3.10-1_amd64.changes
dpkg-genchanges: info: including full source code in upload
 dpkg-source --after-build node-types-jquery-3.3.10
dpkg-buildpackage: info: full upload (original source is included)
dpkg-buildpackage: warning: not signing UNRELEASED build; use --force-sign to override
dh clean
   dh_clean

Remember, your new source directory is @types/jquery/node-types-jquery-3.3.10

This is not a crystal ball, so please take a look at auto-generated files.

You may want fix first these issues:

@types/jquery/node-types-jquery-3.3.10/debian/control:Description: FIX_ME write the Debian package description
@types/jquery/node-types-jquery-3.3.10/debian/copyright:Upstream-Contact: FIX_ME upstream author
@types/jquery/node-types-jquery-3.3.10/debian/copyright:Copyright: 2018, FIX_ME upstream author
@types/jquery/node-types-jquery_itp.mail:Subject: ITP: node-types-jquery -- FIX_ME write the Debian package description
@types/jquery/node-types-jquery_itp.mail:  Upstream Author : FIX_ME upstream author
@types/jquery/node-types-jquery_itp.mail:  Description     : FIX_ME write the Debian package description
@types/jquery/node-types-jquery_itp.mail: FIX_ME: This ITP report is not ready for submission, until you are
@types/jquery/node-types-jquery_itp.mail:FIX_ME: Explain why this package is suitable for adding to Debian. Is
@types/jquery/node-types-jquery_itp.mail:FIX_ME: Explain how you intend to consistently maintain this package

*** Warning ***
Using fakeupstream to download npm dist tarballs, because upstream
git repo is missing tags. Its better to ask upstream to tag their releases
instead of using npm dist tarballs as dist tarballs may contain pre built files
and may not include tests.


```

File: npm2deb/utils.py (hunk 1 above)
Comment (pravi) [reply npm2deb/utils.py]:
```
+ npm2deb view babel-core
npm reports errors about @babel-core module:
npm ERR! code EINVALIDPACKAGENAME
npm ERR! Invalid package name "@babel-core": name can only contain URL-friendly characters
{
  "error": {
    "code": "EINVALIDPACKAGENAME",
    "summary": "Invalid package name \"@babel-core\": name can only contain URL-friendly characters",
    "detail": ""
  }
}
```

**Comment by shanavas786** [comment]
`parse_name` now fails for package names without version or scope

```
def parse_name(name):
    parts = name.rpartition('/')
    scope = parts[0] + parts[1]
    name_version = parts[2].partition('@')
    return scope + name_version[0], name_version[2]
```

This works for the following cases
```
parse_name("package") => ("package", "")
parse_name("package@version") => ("package", "version")
parse_name("@scope/package") => ("@scope/package", "")
parse_name("@scope/package@version") => ("@scope/package", "version")
```

**Comment by shanavas786** [comment]
We have to support the following formats 

```
@scope/package
package
@scope/package@version
package@version
```

**Comment by pravi** [comment]
@shanavas786 Tested OK. @juggernaut451 thanks!

File: npm2deb/utils.py (hunk 1 above)
Comment (juggernaut451) [reply npm2deb/utils.py]: changes made

**Comment by juggernaut451** [comment]
@shanavas786 can you please have a look at test cases nwo

**Pull request merged**